### PR TITLE
feat/ci/action: add gh-pages deployment job

### PR DESCRIPTION
**Pull request by drahnr**
Derived from the polkadot guide.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -297,3 +297,54 @@ jobs:
           hub release create -m $tag_name $tag_name $(for f in cachepot-*.tar.gz*; do echo "-a $f"; done)
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
```
Comment (Xanewok) [inline .github/workflows/ci.yml:300]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -297,3 +297,54 @@ jobs:
...
+        sudo apt install -y graphviz
+        cargo install mdbook-graphviz
+
+    - name: Setup mdbook-mermaid
+      run: |
+        cargo install mdbook-mermaid
+
+    - name: Setup mdbook-linkcheck
+      run: |
+        cargo install mdbook-linkcheck
```
Comment (Xanewok) [inline .github/workflows/ci.yml:340]: This seems to be duplicated

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -297,3 +297,54 @@ jobs:
...
+
+
+  deploy-gh-pages:
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout local repo
+      uses: actions/checkout@v2
+
+    - name: Clone cachepot repo
```
Comment (Xanewok) [inline .github/workflows/ci.yml:309]: Do we need to double-clone the same repository? Can't we run `mdbook build docs/` from the current directory?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -297,3 +297,54 @@ jobs:
...
+        toolchain: stable
+
+    - name: Setup mdbook-graphviz
+      run: |
+        sudo apt install -y graphviz
+        cargo install mdbook-graphviz
+
+    - name: Setup mdbook-mermaid
+      run: |
+        cargo install mdbook-mermaid
```
Comment (Xanewok) [inline .github/workflows/ci.yml:336]: question: FWIU mermaid and graphviz are two ways to visualize graphs - do we need to eventually settle on one for consistency or do we want to use both?

**Review by Xanewok — CHANGES_REQUESTED** [review]
There seems to be some left-overs from adapting Polkadot guide and I'm not sure these should be here, other than that the flow looks good =)

File: .github/workflows/ci.yml (hunk 4 above)
Comment (drahnr) [reply .github/workflows/ci.yml:335]: We don't know at this point, and both are useful and suffer a slightly different purpose, depending on the level of control that is needed. Right now we are not using either, but I don't want to revisit the mdbook/ci topic anytime soon, but we should formalize the flow graphs soon and I'd like to keep the choices open.

**Comment by drahnr** [comment]
Moved the pages to a separate yml file.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -296,4 +296,4 @@ jobs:
           tag_name=${GITHUB_REF#refs/tags/}
           hub release create -m $tag_name $tag_name $(for f in cachepot-*.tar.gz*; do echo "-a $f"; done)
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Xanewok) [inline .github/workflows/ci.yml:299]: micro nitpick: let's keep the trailing EOL

**Review by Xanewok — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 5 above)
Comment (drahnr) [reply .github/workflows/ci.yml:299]: Those whitespaces are getting the best of me :upside_down_face:

**Comment by Xanewok** [comment]
> peaceiris/actions-mdbook@v1, actions-rs/toolchain@v1, and peaceiris/actions-gh-pages@v3 are not allowed to be used in paritytech/cachepot. Actions in this workflow must be: within a repository owned by paritytech, verified in the GitHub Marketplace or match the following: Label305/AutoRebase@*, action-rs/rust-toolchain@*, tibdex/github-app-token@*.

That's something that we'll need to be configured probably after it's merged

**Comment by drahnr** [comment]
> > peaceiris/actions-mdbook@v1, actions-rs/toolchain@v1, and peaceiris/actions-gh-pages@v3 are not allowed to be used in paritytech/cachepot. Actions in this workflow must be: within a repository owned by paritytech, verified in the GitHub Marketplace or match the following: Label305/AutoRebase@_, action-rs/rust-toolchain@_, tibdex/github-app-token@*.
> 
> That's something that we'll need to be configured probably after it's merged

Added them.

**Pull request merged**